tools/add_triage_signature.py

```diff
@@ -189,94 +196,126 @@ def get_mustgather(triage_logs_tar):
+        self.should_reevaluate = dry_run_file
+        self.issue_key = issue_key
 
-    def update_ticket(self, url, issue_key, should_update=False):
+    def process_ticket(self, url, issue_key):
         try:
-            self._update_ticket(url, issue_key, should_update=should_update)
+            self._process_ticket(self._logs_url_to_api(url), issue_key)
         except FailedToGetMetadataException as e:
             browse_url = get_ticket_browse_url(issue_key)
-            logger.error(
-                f"Error getting metadata for {browse_url} at {url}: {e.__cause__}, it may have been deleted"
-            )
+            logger.error(f"Error getting metadata for {browse_url} at {url}: {e.__cause__}, it may have been deleted")
         except FailedToGetLogsTarException as e:
             if e.__cause__.response.status_code == 404:
                 logger.warning(
-                    f"{get_ticket_browse_url(issue_key)} doesn't have a log tar, skipping {type(self).__name__}")
+                    f"{get_ticket_browse_url(issue_key)} doesn't have a log tar, skipping {type(self).__name__}"
+                )
                 return
             raise
         except Exception:
             logger.exception("error updating ticket %s", issue_key)
 
     @abc.abstractmethod
-    def _update_ticket(self, url, issue_key, should_update=False):
+    def _process_ticket(self, url, issue_key):
         pass
 
-    def _add_signature_label(self, key, labels):
-        issue_labels = self._jclient.issue(key).fields.__dict__[custom_field_name(CF_FUNCTION_IMPACT)]
-        if issue_labels is None:
-            issue_labels = []
-        is_changed = False
-        for label in labels:
-            if label not in issue_labels:
-                is_changed = True
-                issue_labels.append(label)
-        if is_changed:
-            self._update_fields(key, {custom_field_name(CF_FUNCTION_IMPACT): issue_labels})
+    def _add_signature_function_impact(self, issue_key, labels_to_add):
+        ticket_existing_function_impact_labels = (
+            self._jira_client.issue(issue_key).fields.__dict__[custom_field_name(CUSTOM_FIELD_FUNCTION_IMPACT)] or []
+        )
+
+        new_labels = [label for label in labels_to_add if label not in ticket_existing_function_impact_labels]
+
+        if len(new_labels) != 0:
+            self._update_fields(
+                issue_key,
+                {custom_field_name(CUSTOM_FIELD_FUNCTION_IMPACT): ticket_existing_function_impact_labels + new_labels},
+            )
+
+    def _add_signature_labels(self, issue_key, labels_to_add):
+        ticket_existing_labels = self._jira_client.issue(issue_key).fields.__dict__[FIELD_LABELS] or []
+        new_labels = [label for label in labels_to_add if label not in ticket_existing_labels]
+
+        if len(new_labels) != 0:
+            self._update_fields(
+                issue_key,
+                {FIELD_LABELS: ticket_existing_labels + new_labels},
+            )
```

Comment:
I think this could be merged into one function that would accept an additional parameter, in this case `FIELD_LABELS` or `custom_field_name(CUSTOM_FIELD_FUNCTION_IMPACT)`. But feel free to keep 2 separate if you think it's easier to digest